nix-store.el

```diff
@@ -206,6 +206,23 @@ It uses \\[nix-store-show-path] to display the store path."
   (interactive)
   (nix-store-show-path (nix-store-path-at-point)))
 
+(defun nix-store-show-log ()
+  "Opens the log file for the derivation of the nix-store path."
+  (interactive)
+  (let ((drv-name (when-let*
+		      ((drv-name (nix-store-path-derivers nix-buffer-store-path))
+		       (drv-name (car drv-name)))
+		    (file-relative-name drv-name nix-store-dir))))
+    (if (not drv-name)
+	(message "This store path has no associated derivation.")
+      (let ((log-file (format "%s/log/nix/drvs/%s/%s.bz2"
+                              nix-state-dir
+                              (substring drv-name 0 2)
+			      (substring drv-name 2))))
```

Comment:
Note that this is slightly different to the `nix-log` function. The `nix-log` function uses:

```elisp
(format "%s/log/nix/drvs/%s/%s.bz2"
                           nix-state-dir
                           (substring drv-name 0 2) drv-name)
```

I haven't yet tried out `nix-log`, so am unsure why this is. Possibly the structure of the nix log directory changed to remove the first two characters of `drv-name`?

Reply:
yeah, I'm not sure if that changed or it's a mistake.

Reply:
I updated it so they both use a utility function so we can correct it for nix-log too.